### PR TITLE
Add configurability for attribute sorting

### LINKED ISSUE #160: Add attribute sorting for items
When adding items with a lot of attributes the default alphabetical sort is not always ok.

A sorting attribute would address this like:

```
.. item:: FMEA_FUNCTION_SYSTEM The system must work
    :attribute_sort: action status 
    :action: We will make the system work
    :status: approved
```

The ones not mentioned could be left  alphabetical after the ones mentioned

### PR DESCRIPTION
**Pull request by JasperCraeghs**
Resolves #160 

- [x] Implemented
- [x] Manually tested
- [x] Added documentation
- [x] Added automated tests

Example usage:
```
.. attribute-sort::
    :filter: RQT-
    :sort: status value aspice
```

### CODE REVIEW

File: doc/design.rst [hunk 1]
```diff
@@ -350,6 +359,11 @@ Design for mlx.traceability
 
     An overview table of the attribute values for documentation parts shall be generated.
 
+.. item:: DESIGN-ATTRIBUTE_SORT Custom sorting of items' attributes
+    :depends_on: DESIGN-ATTRIBUTES
+
+    The plugin shall have a directive that allows configurability of the order of items' attributes.
```
Comment (Letme) [inline doc/design.rst]: Shall=has as these are not requirements so describe what is done.

Comment (Letme) [reply doc/design.rst]: Maybe alao add how sorting and filtering are working?

Comment (Letme) [reply doc/design.rst]: Nevermind last comment= i hace seen the details in usage

Comment (JasperCraeghs) [reply doc/design.rst]: Fixed design items in 8f2f510a77736abca463ea379c69ee2d5bd5238e

**Pull request merged**